opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -1300,13 +1292,21 @@ namespace Opm
             this->operability_status_.obey_bhp_limit_with_thp_limit = (*obtain_bhp >= bhp_limit);
 
             const double thp_limit = this->getTHPConstraint(summaryState);
-            if (*obtain_bhp < thp_limit) {
+            const double temp = *obtain_bhp < thp_limit;
+            if (this->isProducer() && temp < 0.) {
```

Comment:
This looks very unexpected.  The variable `temp` is of type `double`, is initialised with an expression of type `bool`, and is then subsequently compared to a floating-point constant&mdash;the result of which is always `false` since Boolean values are never less than zero.  If this is in fact intended, I suggest you make an effort to think of a better name than `temp`.

Reply:
Obs, this was not my intention. I will clean it up. Should have doubled checked before converting it from draft.....

Reply:
> Obs, this was not my intention. I will clean it up.

Cool.  I look forward to seeing the revised implementation.

> Should have doubled checked before converting it from draft...

Just in case you don't know, it's possible to go back to draft state by clicking the "convert to draft" link in the upper right-hand corner of this page.

---

![convert-to-draft](https://user-images.githubusercontent.com/1871438/137318198-86ffb26f-d5f0-4289-b53f-2aed266f318c.png)